logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/message/MessageSender.kt

```diff
@@ -300,10 +322,11 @@ internal class MessageSenderImpl internal constructor(
     private suspend fun trySendingProteusEnvelope(
         envelope: MessageEnvelope,
         message: Message.Sendable,
+        messageOption: MessageApi.QualifiedMessageOption,
         messageTarget: MessageTarget
```

Comment:
Not ideal to have both `messageOption` and  `messageTarget` as parameters. So the reason we have both is because `messageTarget` is needed when retrying due missing clients for example to keep original intention of sending to a `.conversation` / `.client`.

One way out of this could be to:

1. Only keep `messageTarget` parameter here
2. Modify `MessageTarget` to capture ignore semantics Message.Conversation(val ignoreSome: List<UserId>?)
3. Map MessageOption inside this function
4. Drop IgnoreSome when re-trying.